### PR TITLE
feat : Added history-management

### PR DESCRIPTION
**Pull request by deepak29-git**
1.Add to history
2 delete from history
3. clear all history

### CODE REVIEW

File: src/Pages/History/History.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { Header } from "../../components/Header/Header";
+import { Sidebar } from "../../components/Sidebar/Sidebar";
+import { useHistory } from "../../Context/history-context";
+import { clearAllWatchHistory } from "../../Utility/clear-all-watch-history";
+import { deleteFromHistory } from "../../Utility/delete-from-history";
+import "../History/History.css";
+export const History = () => {
+  const { history,setHistory } = useHistory();
+  console.log(history, "history");
```
Comment (prathmeshjagtap) [inline src/Pages/History/History.jsx:9]: remove console.log

File: src/Utility/clear-all-watch-history.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+export const clearAllWatchHistory=async(setHistory)=>{
+    try{
+        const {data}=await axios.delete("/api/user/history/all",{
+            headers:{
+                authorization:getToken()
+            }
+        })
+        setHistory(data.history)
+        console.log(data.history)
```
Comment (prathmeshjagtap) [inline src/Utility/clear-all-watch-history.js:12]: remove console.log

**Review by prathmeshjagtap — APPROVED** [review]
code is Clean and I liked that you have made separate functions to get history data

**Review by sharathnair9999 — APPROVED** [review]

File: src/Pages/History/History.css [hunk 3]
```diff
@@ -0,0 +1,19 @@
+.grid-container {
+  grid-gap: 15px;
+}
+
+.video-container {
+  height: 350px;
```
Comment (shivanigangadharan) [inline src/Pages/History/History.css:6]: Can keep this in rem

File: src/Pages/VideoListing/VideoListing.css [hunk 4]
```diff
@@ -100,3 +105,26 @@ label {
...
+  .video-listing-main {
+    margin-top: 8rem;
+  }
+}
+
+@media only screen and (max-width: 490px) {
+  .video-listing-main {
+    margin-top: 12rem;
+  }
+} */
```
Comment (shivanigangadharan) [inline src/Pages/VideoListing/VideoListing.css:130]: Can removed comments if not reqd anymore

File: src/components/Sidebar/Sidebar.jsx [hunk 5]
```diff
@@ -1,9 +1,11 @@
 import "../Sidebar/Sidebar.css";
 import { Link } from "react-router-dom";
+import { useData } from "../../Context/data-context";
 export const Sidebar = () => {
+  const {sidebar}=useData()
   return (
     <div>
-      <aside style={{margin:"0"}} className="video-library-sidebar">
+      <aside style={{position:"fixed",left:sidebar,transition:"all 0.5s ease-in-out"}}  className="video-library-sidebar">
```
Comment (shivanigangadharan) [inline src/components/Sidebar/Sidebar.jsx:8]: can avoid inline styling and add these styles in class

**Review by shivanigangadharan — APPROVED** [review]
Overall code looks good to merge, I like that you've added custom routes as well.

File: src/components/Sidebar/Sidebar.jsx (hunk 5 above)
Comment (deepak29-git) [reply src/components/Sidebar/Sidebar.jsx:8]: ok

File: src/Pages/History/History.css (hunk 3 above)
Comment (deepak29-git) [reply src/Pages/History/History.css:6]: ok

File: src/Utility/clear-all-watch-history.js (hunk 2 above)
Comment (deepak29-git) [reply src/Utility/clear-all-watch-history.js:12]: ok

File: src/Pages/History/History.jsx (hunk 1 above)
Comment (deepak29-git) [reply src/Pages/History/History.jsx:9]: ok

**Pull request merged**